azure/scope/managedcontrolplane.go

```diff
@@ -82,6 +84,10 @@ func NewManagedControlPlaneScope(ctx context.Context, params ManagedControlPlane
 		}
 	}
 
+	if params.Cache == nil {
+		return nil, errors.New("failed to generate new scope from nil Cache")
```

Comment:
nit: `failed to generate new scope due to nil Cache`

Reply:
no longer returning error for nil Cache